clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -345,7 +374,9 @@ def DataMemberAttr : CIR_Attr<"DataMember", "data_member",
   let parameters = (ins AttributeSelfTypeParameter<
                             "", "mlir::cir::DataMemberType">:$type,
                         OptionalParameter<
-                            "std::optional<size_t>">:$memberIndex);
+                            "std::optional<unsigned>">:$memberIndex,
```

Comment:
While you're here, snake_case for all operands plz!

Reply:
Updated.